pkg/operators/v1alpha1/subscription_types.go

```diff
@@ -29,13 +29,13 @@ const (
 
 // SubscriptionSpec defines an Application that can be installed
 type SubscriptionSpec struct {
-	CatalogSource          string             `json:"source"`
-	CatalogSourceNamespace string             `json:"sourceNamespace"`
-	Package                string             `json:"name"`
-	Channel                string             `json:"channel,omitempty"`
-	StartingCSV            string             `json:"startingCSV,omitempty"`
-	InstallPlanApproval    Approval           `json:"installPlanApproval,omitempty"`
-	Config                 SubscriptionConfig `json:"config,omitempty"`
+	CatalogSource          string              `json:"source"`
+	CatalogSourceNamespace string              `json:"sourceNamespace"`
+	Package                string              `json:"name"`
+	Channel                string              `json:"channel,omitempty"`
+	StartingCSV            string              `json:"startingCSV,omitempty"`
+	InstallPlanApproval    Approval            `json:"installPlanApproval,omitempty"`
+	Config                 *SubscriptionConfig `json:"config,omitempty"`
```

Comment:
This change is not directly related to bug fix with overriding resources. However it seems that the intent is for `spec.config` to be optional, so it seems like a reasonable time to make this change as well.